src/dos/cdrom.cpp

```diff
@@ -18,6 +18,217 @@
 
 #include "cdrom.h"
 
+#include "../libs/sdlcd/SDL_cdrom.c"
+#if defined(WIN32)
+#define SDL_CDROM_WIN32
+#include "../libs/sdlcd/win32/SDL_syscdrom.c"
+#elif defined(LINUX)
+#define SDL_CDROM_LINUX
+#include "../libs/sdlcd/linux/SDL_syscdrom.c"
+#elif defined(MACOSX) && defined(__clang__)
+#define SDL_CDROM_MACOSX
+#include "../libs/sdlcd/macosx/SDL_syscdrom.c"
+#include "../libs/sdlcd/macosx/AudioFilePlayer.c"
+#include "../libs/sdlcd/macosx/AudioFileReaderThread.c"
+#include "../libs/sdlcd/macosx/CDPlayer.c"
+#include "../libs/sdlcd/macosx/SDLOSXCAGuard.c"
+#else
+#define SDL_CDROM_DUMMY
+#include "../libs/sdlcd/dummy/SDL_syscdrom.c"
+#endif
```

Comment:
I've never been a fan of directly including C files unless absolutely necessary.

If possible, `sdlcd` should be compiled separately and linked by the build system, like most other code is, and then `#include SDL_syscdrom.h"`

I notice each platform specific version of `SDL_syscdrom.c` is guarded by `SDL_CDROM_<platform>` define.

Reply:
Thanks for the suggestion. Will change the build file instead.

Reply:
I have converted direct file conversions to build systems (Meson and Visual Studio). It took some efforts, but appears to work.